compiler/AST/symbol.cpp

```diff
@@ -144,10 +144,15 @@ static inline void verifyInTree(BaseAST* ast, const char* msg) {
 }
 
 void Symbol::verify() {
-  if (defPoint && !defPoint->parentSymbol && !toModuleSymbol(this))
-    INT_FATAL(this, "Symbol::defPoint is not in AST");
-  if (defPoint && this != defPoint->sym)
-    INT_FATAL(this, "Symbol::defPoint != Sym::defPoint->sym");
+  if (defPoint) {
+    if (!defPoint->parentSymbol && this != rootModule)
+      INT_FATAL(this, "Symbol::defPoint is not in AST");
+    if (this != defPoint->sym)
+      INT_FATAL(this, "Symbol::defPoint != Sym::defPoint->sym");
+  } else {
+    if (this != rootModule)
+      INT_FATAL(this, "Symbol without a defPoint");
+  }
```

Comment:
I agree this version is clearer. However it adds a branch + memory access to execute "this == rootModule" for each symbol, whereas my version executes it only for the rootModule symbol.

You are correct, rootModule->defPoint is NULL. So indeed there is no checking to be performed in that case... although I would throw in the check that it is actually NULL - to catch an unlikely bug when it is not.